bigcode_eval/wx_ai.py

```diff
@@ -0,0 +1,154 @@
+            "top_p": args.top_p,
+            "top_k": args.top_k,
+            "max_new_tokens": args.max_new_tokens,
+            "min_new_tokens": args.min_new_tokens,
+            "length_penalty": args.length_penalty,
+            "stop_sequences": args.stop_sequences,
+        }
+
+    def initialize_wx_model(self, model_name: str) -> ModelInference:
+        return ModelInference(
+            model_id=model_name,
+            api_client=self.client,
+        )
+
+    @staticmethod
+    def create_prompt_from_dict(
+        prompt_content: dict[str, str],
+        prefix: str,
+        instruction_tokens: Optional[list[str]],
+    ) -> str:
+        if all(key in ("instruction", "context") for key in prompt_content):
+            user_token, end_token, assistant_token = "", "", "\n"
+            if instruction_tokens:
+                user_token, end_token, assistant_token = instruction_tokens
+
+            return "".join(
+                (
+                    prefix,
+                    user_token,
+                    prompt_content["instruction"],
+                    end_token,
+                    assistant_token,
+                    prompt_content["context"],
+                )
+            )
+
+        # Infilling mode is only used for particular models, none of which is available in wx.
+        elif all(key in ("prefix", "suffix") for key in prompt_content):
+            return f"{prefix}{prompt_content['prefix']}{prompt_content['suffix']}"
+
+        else:
+            raise ValueError(
+                f"Unsupported prompt format: '{prompt_content}'."
+            )
+
+    def prepare_inputs(
+        self,
+        dataset: Dataset,
+        task: Task,
+        prefix: str,
+        instruction_tokens: Optional[list[str]],
+    ) -> list[str]:
+        sample = task.get_prompt(dataset[0])
+
+        assert isinstance(sample, (str, dict)), (
+            f"Prompt must be either a string, or a dictionary. "
+            f"However, '{sample}' was given, which is "
+            f"of type: '{type(sample)}'."
+        )
+
+        return [
+            prefix + task.get_prompt(instance)
+            if isinstance(sample, str)
+            else self.create_prompt_from_dict(
+                task.get_prompt(instance), prefix, instruction_tokens
+            )
+            for instance in dataset
+        ]
+
+    def infer(
+        self,
+        dataset: Dataset,
+        task: Task,
+        args: Namespace,
+        prefix: str = "",
+        postprocess: bool = True,
+    ) -> list[list[str]]:
+        gen_params = self.prepare_wx_gen_params(args)
+
+        model = self.initialize_wx_model(args.model)
+
+        prompts = self.prepare_inputs(
+            dataset, task, prefix, args.instruction_tokens
+        )
+
+        predictions = [
+            [result["results"][0]["generated_text"]]
+            for result in
+            model.generate(
+                prompt=prompts,
```

Comment:
Maybe we should limit number of prompts:
```suggestion
                prompt=prompts[: args.limit],
```